source/plugin/permissions.rst

```diff
@@ -228,14 +228,65 @@ mechanisms themselves.
 Contexts
 ========
 
-If you consider each permission to a privilege or ability to be able to do something, a Context is the circumstances
+If you consider each permission to a privilege or ability to be able to do something, a ``Context`` is the circumstances
 where that privilege is usable.
 
-You might want to give a Subject permission to do something, but only when the Subject is in a certain world, or in a
-certain Region.
+You might want to give a ``Subject`` permission to do something, but only when the ``Subject`` is in a certain world,
+or in a certain region.
 
-Contexts are accumulated by a Subject, and are then used by the PermissionService to decide if the Subject has a
-privilege or not.
+Contexts are accumulated by a ``Subject``, and are then used by the ``PermissionService`` to decide if the ``Subject``
+has a privilege or not.
 
 Sponge provides some contexts by default, but it is generally down to other plugins to provide additional contexts to
-the PermissionService, through a ContextCalculator.
\ No newline at end of file
+the PermissionService, through a ``ContextCalculator``.
+
+When creating contexts for your own plugin please try to avoid conflicts with other plugins (ex by choosing a unique
+context key) unless these contexts are meant to be shared.
+
+.. note::
+    
+    Please make sure that your ``ContextCalculator`` responds as fast as possible as it will get called frequently.
+
+Example
+~~~~~~~
+
+Your ``ContextCalculator`` may look like this:
+
+.. code-block:: java
+
+    public class MyArenaPlugin implements ContextCalculator<Subject> {
+    
+        private final Context IN_ANY_ARENA = new Context("myArenaPlugin", "*");
+    
+        @Override
+        public void accumulateContexts(Subject subject, Set<Context> accumulator) {
```

Comment:
Ok, better example.

You want a permission to only apply on the "world_nether" world, (I know Sponge gives them DIM-x names, whatever, just an example.)

The key world be "world", the value would be "world_nether". 

There is no such thing as "wildcard" contexts, it's down to the plugin who provides the calculator, or the plugin implementing permissions.

My point is, instead of: key="in_arena", value="*", it would make more sense to have key="in_arena", value="true", or have the value equal the name of the arena.

There's no concept of wildcard Contexts, and especially in the documentation, it's important to show that Contexts are a collection of key value pairs, not a collection of keys.

(I actually think the world example is the easiest to understand, however I understand you're trying to document the calculator here, not the concept of Contexts)